lib/features/public_asset/domain/usecase/clean_up_public_assets_interactor.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:tmail_ui_user/features/public_asset/domain/model/public_assets_in_identity_arguments.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/state/clean_up_public_assets_state.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/usecase/add_identity_to_public_assets_interactor.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/usecase/delete_public_assets_interactor.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/usecase/remove_identity_from_public_assets_interactor.dart';
+import 'package:tmail_ui_user/features/public_asset/presentation/public_asset_controller.dart';
+
+class CleanUpPublicAssetsInteractor {
+  CleanUpPublicAssetsInteractor(
+    this.removeIdentityFromPublicAssetsInteractor,
+    this.deletePublicAssetsInteractor,
+    this.addIdentityToPublicAssetsInteractor);
+
+  final RemoveIdentityFromPublicAssetsInteractor removeIdentityFromPublicAssetsInteractor;
+  final DeletePublicAssetsInteractor deletePublicAssetsInteractor;
+  final AddIdentityToPublicAssetsInteractor addIdentityToPublicAssetsInteractor;
+
+  Stream<Either<Failure, Success>> execute(
+    Session session,
+    AccountId accountId,
+    {
+      required IdentityId identityId,
+      required IdentityActionType identityActionType,
+      required PublicAssetsInIdentityArguments publicAssetsInIdentityArguments,
+    }
+  ) async* {
+    try {
+      yield Right(CleaningUpPublicAssetsState());
+
+      final publicAssetIds = _filterPublicAssetsIdsForCleanUp(
+        identityId,
+        identityActionType,
+        publicAssetsInIdentityArguments);
+
+      await Rx.merge<Either<Failure, Success>>([
+        if (publicAssetIds.publicAssetsIdsToBeDereferenced.isNotEmpty)
+          removeIdentityFromPublicAssetsInteractor.execute(
+            session,
+            accountId,
+            identityId: identityId,
+            publicAssetIds: publicAssetIds.publicAssetsIdsToBeDereferenced),
+        if (publicAssetIds.publicAssetsIdsToBeDestroyed.isNotEmpty)
+          deletePublicAssetsInteractor.execute(
+            session,
+            accountId,
+            publicAssetIds: publicAssetIds.publicAssetsIdsToBeDestroyed),
+        if (publicAssetIds.publicAssetsIdsToBeReferenced.isNotEmpty)
+          addIdentityToPublicAssetsInteractor.execute(
+            session,
+            accountId,
+            identityId: identityId,
+            publicAssetIds: publicAssetIds.publicAssetsIdsToBeReferenced),
+      ]).last;
+
+      yield Right(CleanUpPublicAssetsSuccessState());
+
+    } catch (exception) {
+      logError('CleanUpPublicAssetsInteractor::execute():error: $exception');
+      yield Left(CleanUpPublicAssetsFailureState(exception: exception));
+    }
+  }
+
+  ({
+    List<PublicAssetId> publicAssetsIdsToBeDestroyed,
+    List<PublicAssetId> publicAssetsIdsToBeReferenced,
+    List<PublicAssetId> publicAssetsIdsToBeDereferenced, 
+  }) _filterPublicAssetsIdsForCleanUp(
```

Comment:
IMO, we also should write unit test for it

Reply:
He did for clean up interactor. all tests for this belong to CleanUp test. no need to write test for private function